File: java/src/main/java/com/alxarafe/app/domain/model/Greeting.java
package com.alxarafe.app.domain.model;

import java.time.Instant;

public final class Greeting {
    private final String id;
    private final String message;
    private final Instant createdAt;

    public Greeting(String id, String message, Instant createdAt) {
        this.id = id;
        this.message = message;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public String getMessage() {
        return message;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }
}

File: php/src/Domain/Model/Greeting.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Domain\Model;

final readonly class Greeting
{
    public function __construct(
        private string $id,
        private string $message,
        private \DateTimeImmutable $createdAt,
    ) {
    }

    public function getId(): string
    {
        return $this->id;
    }

    public function getMessage(): string
    {
        return $this->message;
    }

    public function getCreatedAt(): \DateTimeImmutable
    {
        return $this->createdAt;
    }
}
